test/test_common/environment.cc

```diff
@@ -375,4 +412,25 @@ void TestEnvironment::setRunfiles(Runfiles* runfiles) { runfiles_ = runfiles; }
 
 Runfiles* TestEnvironment::runfiles_{};
 
+AtomicFileUpdater::AtomicFileUpdater(const std::string& filename)
+    : link_(filename), new_link_(absl::StrCat(filename, ".new")),
+      target1_(absl::StrCat(filename, ".target1")), target2_(absl::StrCat(filename, ".target2")),
+      use_target1_(true) {
+  unlink(link_.c_str());
+  unlink(new_link_.c_str());
+  unlink(target1_.c_str());
+  unlink(target2_.c_str());
```

Comment:
This seems odd? Don't you want these things to exist outside of calling `update`? (Edit: I see that you copied this, though it still doesn't make sense to me. Can you audit and/or add more class comments to explain what this is doing)?

Reply:
From the test environment perspective, there is no reason to support a pre-existing resource at this time. Further thoughts on purpose and scope can be found in the original discussion at https://github.com/envoyproxy/envoy/pull/4298
If a preexisting file were to be supported, it would be renamed to target2 and replaced by a symlink; if a preexisting symlink were to be supported it could be preserved on creation, I suppose. But it's all out of scope of this change, as it simply moves as-implemented from TestUtility to TestEnvironment and appears to be working on master as-is.

Reply:
OK, please add more comments on the class itself on what this can currently be used for and the fact that it will clear any existing file, etc.